File: src/main/java/com/github/nsc/de/shake/interpreter/values/StringValue.java
package com.github.nsc.de.shake.interpreter.values;

public class StringValue implements InterpreterValue {

    private final String value;

    public StringValue(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public StringValue add(InterpreterValue v) {
        return new StringValue(this.value + v.toJava());
    }

    /**
     * Get the java-representation of the {@link InterpreterValue}
     *
     * @return the java-representation of the {@link InterpreterValue}
     * @author <a href="https://github.com/nsc-de">Nicolas Schmidt &lt;@nsc-de&gt;</a>
     */
    @Override
    public String toJava() {
        return value;
    }

    /**
     * Returns the name of the type of {@link InterpreterValue} (To identify the type of value)
     *
     * @return the name of the {@link InterpreterValue}
     * @author <a href="https://github.com/nsc-de">Nicolas Schmidt &lt;@nsc-de&gt;</a>
     */
    @Override
    public String getName() {
        return "string";
    }

    @Override
    public String toString() {
        return String.format("\"%s\"", value);
    }
}

File: shake/compiler/interpreter/src/commonMain/kotlin/io/github/shakelang/shake/interpreter/values/StringValue.kt
package com.shakelang.shake.interpreter.values

class StringValue(val value: String) : InterpreterValue {

    override fun add(v: InterpreterValue): StringValue = StringValue(value + v.toJava())

    /**
     * Get the java-representation of the [InterpreterValue]
     *
     * @return the java-representation of the [InterpreterValue]
     */
    override fun toJava(): String = value

    /**
     * Returns the name of the type of [InterpreterValue] (To identify the type of value)
     *
     * @return the name of the [InterpreterValue]
     */
    override val name: String get() = "string"

    override fun toString(): String = "\"$value\""
}